Fix building problem - MPLAB

When building the 2FOC.V3 project, the following error appears:

```../src/ecan.c:827:37: error: conflicting types for 'int8_t'```

 It is a problem related to the new compiler version. The old version contains this line:

https://github.com/robotology/icub-firmware/blob/b6c0a716bb75105fd1d62cfce6ae05c7d649df9b/emBODY/eBcode/arch-dspic/board/2foc/appl/2FOC-V3/src/ecan.c#L827

that causes that error.

The fix is:

https://github.com/isorrentino/icub-firmware/blob/afd340abb343c5d5502970961266feaac327dd8a/emBODY/eBcode/arch-dspic/board/2foc/appl/2FOC-V3/src/ecan.c#L1135

--- a/emBODY/eBcode/arch-dspic/board/2foc/appl/2FOC-V3/src/ecan.c
+++ b/emBODY/eBcode/arch-dspic/board/2foc/appl/2FOC-V3/src/ecan.c
@@ -6,7 +6,7 @@
 
 #include <p33FJ128MC802.h>
 #include <string.h>
-
+#include "stdint.h"
 #include "ecan.h"
 #include "system.h"
 #include "faults.h"
@@ -823,15 +823,6 @@ void __attribute__((interrupt, no_auto_psv)) _DMA2Interrupt(void)
 
 
 //VALE
-
-typedef     char                    int8_t;
-typedef     unsigned char           uint8_t;
-typedef     signed int              int16_t;
-typedef     unsigned int            uint16_t;
-typedef     signed long             int32_t;
-typedef     unsigned long           uint32_t;    
-typedef     signed long long        int64_t;
-typedef     unsigned long long      uint64_t; 
 
     
 extern void hal_can_receptionfilter_set(unsigned char mask_num, unsigned long mask_val, unsigned char identifier_num,

--- a/emBODY/eBcode/arch-dspic/board/2foc/appl/2FOC-V3/src/ecan.c
+++ b/emBODY/eBcode/arch-dspic/board/2foc/appl/2FOC-V3/src/ecan.c
@@ -6,7 +6,7 @@
 
 #include <p33FJ128MC802.h>
 #include <string.h>
-#include "stdint.h"
+#include <stdint.h>
 #include "ecan.h"
 #include "system.h"
 #include "faults.h"
@@ -81,7 +81,7 @@ void ECANClkInit(void)
   // Phase Segment 2 = 6Tq
   // Propagation Delay = 5Tq
   // Sync Segment = 1Tq
-  // CiCFG1<BRP> =(FCAN /(2 ×N×FBAUD))– 1
+  // CiCFG1<BRP> =(FCAN /(2 Ã—NÃ—FBAUD))â€“ 1
   // Bit rate of 1Mbps
   C1CFG1bits.BRP = BRP_VAL ;
   // Phase Segment 1 time is 8 Tq

